### PR TITLE
TRD: Grab CCDB Chamber Status for plots

### PR DESCRIPTION
**Pull request by leo-barreto**
## Main Goal
To add the information of Chamber Status, saved in CCDB, to the following QC plots: `trackletsperHC2D`, `TrackletsPerMCM_Layer[0..5]` and `DigitsPerLayer_[0..5]`.

Half chamber status is showed as a black box with a X in it, whenever chamber status is considered **BAD**, i.e. different from CONFIGURED "3" and EMPTY/no data from DCS "0". 

See plots as examples (for simulated data).
trackletsperHC2D

TrackletsPerMCM_Layer0

DigitsPerLayer_0

Also, clean up and organization of TrackletsTask and DigitsTask.

---
Relevant JIRA: [TRD 109](https://alice.its.cern.ch/jira/browse/TRD-109)
Relevant O2 PR: [TRD: Implementation of DCS Chamber Status processing and DPs update](https://github.com/AliceO2Group/AliceO2/pull/12319)

### CODE REVIEW

**Comment by martenole** [comment]
Thanks @leo-barreto ! Did not go through it in detail yet. But could you please also cross out the chambers which did not send any data to the DCS processor? Possibly in another color (gray?), but if that's not easily possible then black also works.
And thanks for the cleanup! To improve the readability and maintainability even more, could you please create a new class, called e.g. TRDHelpers and move all the generic functions which are used both in the tracklets and digits task in there? That reduced the size of the code and if anything is changed in the future one could otherwise easily forget to update both tasks..

File: Modules/TRD/include/TRD/TRDHelpers.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{
+
+class TRDHelpers
+{
+ public:
+  TRDHelpers() = default;
+  ~TRDHelpers() = default;
+
+  void drawTrdLayersGrid(TH2F* hist, int unitspersection);
+  void drawChamberStatusOn2D(std::shared_ptr<TH2F> hist, const std::array<int, o2::trd::constants::MAXCHAMBER>* ptrChamber);
```
Comment (martenole) [inline Modules/TRD/include/TRD/TRDHelpers.h:36]: Can we not use only this method and get rid off `drawChamberStatusOnLayers`?

**Review by martenole — CHANGES_REQUESTED** [review]
Thanks Leonardo, I think this looks already quite good. But while we're doing some cleanup I think it would be good to make the methods in `TRDHelpers` a bit clearer. Sometimes the names are not super descriptive. And if we cannot find a better name we should at least add some comments describing what the method is doing. I know these are all "inherited problems".. But it would be great if we could improve the readability of our code, also for future developers

File: Modules/TRD/src/TRDHelpers.cxx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        int rowMax = (iStack == 2) ? 12 : 16;
+        for (int side = 0; side < 2; ++side) {
+          int det = iSec * 30 + iStack * 6 + iLayer;
+          int hcid = (side == 0) ? det * 2 : det * 2 + 1;
+          int rowstart = iStack < 3 ? iStack * 16 : 44 + (iStack - 3) * 16;                 // pad row within whole sector
+          int rowend = iStack < 3 ? rowMax + iStack * 16 : rowMax + 44 + (iStack - 3) * 16; // pad row within whole sector
+          if (isHalfChamberMasked(hcid, ptrChamber)) {
+            int hcstat = (*ptrChamber)[hcid / 2];
+            std::cout << "Leonardo hcid / 2 " << hcid / 2 << std::endl;
+            std::cout << "Leonardo hcstat " << hcstat << std::endl;
```
[the remark this answers is not in the document]
Comment (leo-barreto) [reply Modules/TRD/src/TRDHelpers.cxx]: My bad! :sweat_smile:

**Review by martenole — APPROVED** [review]
Looks good now! Did you test your latest changes and could you confirm that the DigitsTask and TrackletsTask are producing the objects as you expect them @leo-barreto ? From my side this would then be ready for merging

**Comment by martenole** [comment]
Actually, I see only now that there are conflicts in CMakeLists.txt. Could you please resolve them? In the current maser some classes have been removed.

**Comment by leo-barreto** [comment]
@martenole I did a quick "hand-crafted" chamber status configuration for testing (also uploaded to [ccdb-test](http://ccdb-test.cern.ch:8080/browse/TRD/Calib/DCSDPsFedChamberStatus)). 

It should show, for `i` being the chamber ID:
  - Empty (grey X): `i % 30 == 6`
  - Error (red X): `i` being 42, 121, 420
  - Other masked (black X): `i` between 162 and 179 
  - Everything else: no mask
  
 Current results are: 
trackletsperHC2D_test
TrackletsPerMCM_Layer0_test
DigitsPerLayer_0_test

I had a quick glance at the other plots, and they seem unchanged.

**Thus everything seems okay on my part.**

**Comment by leo-barreto** [comment]
There is an error caught on the automatic build: 

> reco_NOGPU.log-[80011:qc-task-TRD-Tracklets]: [19:10:30][WARN] Did not find object under expected name ccdb_object
> reco_NOGPU.log:[80011:qc-task-TRD-Tracklets]: [19:10:30][ERROR] Exception caught: Couldn't retrieve object corresponding to array<int,540> from TFile

(same thing for qc-task-TRD-Digits)

Maybe this could be fixed with changing the [TRDQC.json](https://github.com/AliceO2Group/QualityControl/blob/64fa7f18968589110b8f2b2b9f34458ba0be86fc/Modules/TRD/TRDQC.json#L60)? "querry" should be changed to the new CCDB path (TRD/Calib/DCSDPsChamberStatus).

@martenole let me know what you think. I'll wait for confirmation before commiting any change to the json file (I don't have it latest version in my branch).

**Review by martenole — APPROVED** [review]
All checks are passing now. From my side this can be merged

**Comment by leo-barreto** [comment]
Everything also good from my side for merge

**Comment by martenole** [comment]
Hi @Barthelemy or @knopers8 could this be merged?

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**